File: network/RequestParser.py
class RequestParser:

    def get_pwid(self, request):
        request_body = request.split("\r\n\r\n")[1]
        id, pw = request_body.split("/")
        return id, pw

    def get_type(self, request):
        type = request.split("\r\n\r\n")[1]
        return type

    # json을 dictionary로 받아와서 문자열("course/material")로 변환
    def documentlist_parser(self, json_documentlist):
            documentlist = ""
            for key, value in json_documentlist.items():
                if isinstance(value, list):
                    for v in value:
                        documentlist += key.strip() + "/" + v + "\n"
                else:
                    documentlist += key.strip() + "/" + value + "\n"

            document_list = documentlist[:-1]
            return document_list

    # missing_document를 list로 만듦.
    def document_parser(self, request):
        documentlist = []
        request_body = request.split("\r\n\r\n")[1]
        data_list = request_body.split("\r\n")
        for data in data_list:
            documentlist.append(data)
        return documentlist

File: src/network/RequestParser.java
package network;

import data.DocumentList;

public class RequestParser {
	
	// 올바른 Response일 때만 로그인 이벤트 처리
	public boolean loginResponseParser(String response) {
		
		if (response.indexOf("HTTP/") != -1) {
			if (response.indexOf("200 OK") != -1) {
				return true;
			}
		}
		return false;
	}
	
	public DocumentList getDocumentResponseParser(String response) {
		DocumentList documentlist = new DocumentList();
		
		if (response.indexOf("HTTP/") != -1) {
			if (response.indexOf("200 OK") != -1) {
				if(response.indexOf("Content-Length: 0") == -1) { //body가 있으면 
					String responsebody  = response.split("\r\n\r\n")[1];
					String[] data_list = responsebody.split("\n");
					for(String data : data_list) {
						String course = data.split("/")[0];
						String material = data.split("/")[1].strip();
						documentlist.addDocument(course, material);
					}
				}
			}
		}
		
		return documentlist;
	}
}
